lucene/core/src/java/org/apache/lucene/index/MergePolicy.java

```diff
@@ -399,6 +427,23 @@ public String segString(Directory dir) {
       }
       return b.toString();
     }
+
+    /**
+     * Waits if necessary for at most the given time for all merges.
+     */
+    boolean await(long timeout, TimeUnit unit) {
+      try {
+        CompletableFuture<Void> future = CompletableFuture.allOf(merges.stream()
```

Comment:
What I learned today: `CompletableFuture.allOf(....)` -- cool, thanks!  So elegant.

Let me try and return the favor:
You can go directly to an array, avoiding toList:
```
CompletableFuture<Void> future = CompletableFuture.allOf(merges.stream()
            .map(m -> m.completable).toArray(CompletableFuture<?>[]::new));
```
BTW IntelliJ pointed this out and had an automatic replacement when I hovered my cursor over "collect".

Reply:
done thanks